source/languages/shell/generate.rb

```diff
@@ -389,31 +378,82 @@ def generateArraySubscriptBracket(bracket)
         }
     ]
     # remove legacy support to fix pattern priorities
-    grammar[:support]["patterns"].pop()
+	grammar[:support]["patterns"].pop()
+
+	### :logical_expression_single, :logical_expression_double ###
+
+	# N.B.: * corresponds to JS's rest operator, which enables us to pack
+	# an arbitrary number of arguments into an array, whereas in JS, we need
+	# to pass an object literal to get named parameters.
+	# In Ruby, we can use double splat operator to pack named parameters
+	# and their values into a hash, as we can do in Perl (%args = @_).
```

Comment:
Sorry for this trivial comment, but before I started working on this script, I merely knew that Ruby is a programming language, so when I learned something new, I write a note for me where I applied it.